extensions/cache/runtime/src/main/java/io/quarkus/cache/Cache.java

```diff
@@ -0,0 +1,56 @@
+package io.quarkus.cache;
+
+import java.util.concurrent.Callable;
+
+import io.quarkus.cache.runtime.caffeine.CaffeineCache;
+import io.smallrye.mutiny.Uni;
+
+/**
+ * This interface can be used to interact with a cache programmatically and store, retrieve or delete cache values.
+ * Most of its operations are reactive.
+ */
+public interface Cache {
+
+    /**
+     * Returns the unique and immutable default key for the current cache. This key is used by the annotations caching API when
+     * a no-args method annotated with {@link CacheResult} or {@link CacheInvalidate} is invoked. It can also be used with the
+     * programmatic caching API.
+     * 
+     * @return default cache key
+     */
+    Object getDefaultKey();
+
+    /**
+     * Returns a lazy asynchronous action that will emit the cache value identified by {@code key}, obtaining that value from
```

Comment:
reading the JavaDoc, I don't know what a value loader is.

Reply:
I'll fix this one when the [callable/supplier/function discussion below](https://github.com/quarkusio/quarkus/pull/8631#discussion_r416407132) comes to a conclusion.

Reply:
Javadoc improved.